### PR TITLE
Soft-Refactor Geiger Counters

### PR DESCRIPTION
**Pull request by VMSolidus**
# Description

This refactors Geiger Counters so that their behavior of "Only making sound to a person holding them" is no longer hardcoded. The GeigerCounterComponent now can define how loud it ticks, how far away people can hear the ticks, and whether it plays only for the person holding it or for anyone nearby. This PR partially fulfills one of the "Nice To Have" features requested for https://github.com/Simple-Station/Einstein-Engines/pull/341 by making it possible to create stationary radiation alarm objects. It also serves as a substantial quality of life improvement for Engineering and Science crew, since it's now possible to place an active Geiger counter in the artifact lab, and then be able to audibly hear if the lab becomes radioactive due to an artifact. 

<details><summary><h1>Media</h1></summary>
<p>

https://github.com/user-attachments/assets/74122135-7345-4995-bb0e-d1216e1d53b6

https://github.com/user-attachments/assets/de79db6f-e1c1-471f-88b5-0a47ff4bfa16

</p>
</details>

# Changelog

:cl:
- add: Geiger Counters other than ones installed in Hardsuits now generate an audible sound when active and exposed to radiation.
- add: Wall mounted geiger counters have been added to the game.

### CODE REVIEW

**Comment by VMSolidus** [comment]
https://github.com/user-attachments/assets/050d001e-ff3d-4dcc-ad5d-fb8f318867dc

And now for Wall Mounted Geiger Counters!

File: Content.Shared/Radiation/Components/GeigerComponent.cs [hunk 1]
```diff
@@ -83,6 +79,31 @@ public sealed partial class GeigerComponent : Component
...
+    ///     Useful for things like hardsuits with integrated geigers. Alternatively, to create stationary radiation alarm objects.
+    /// </summary>
+    [DataField]
+    public bool LocalSoundOnly = false;
+
+    /// <summary>
+    ///     Used to control how loudly the geiger counter ticks.
+    /// </summary>
+    [DataField]
+    public float VolumeParams = -4f;
```
Comment (DangerRevolution) [inline Content.Shared/Radiation/Components/GeigerComponent.cs:94]: Rename all these components to say Parameter

**Review by DangerRevolution — CHANGES_REQUESTED** [review]

**Review by DangerRevolution — APPROVED** [review]

File: Content.Server/Radiation/Systems/GeigerSystem.cs [hunk 2]
```diff
@@ -152,19 +152,23 @@ private void UpdateSound(EntityUid uid, GeigerComponent? component = null)
...
-        var param = sounds.Params.WithLoop(true).WithVolume(-4f);
-
-        component.Stream = _audio.PlayGlobal(sound, session, param)?.Entity;
+        if (component.Sounds.TryGetValue(component.DangerLevel, out var sounds))
+        {
+            var sound = _audio.GetSound(sounds);
+            var param = sounds.Params.WithLoop(true)
+                                     .WithVolume(component.VolumeParameters)
+                                     .WithRolloffFactor(component.RolloffFactorParameters)
+                                     .WithMaxDistance(component.MaxDistanceParameters);
```
Comment (DEATHB4DEFEAT) [inline Content.Server/Radiation/Systems/GeigerSystem.cs:161]:
```suggestion
            var param = sounds.Params.WithLoop(true)
	            .WithVolume(component.VolumeParameters)
	            .WithRolloffFactor(component.RolloffFactorParameters)
	            .WithMaxDistance(component.MaxDistanceParameters);
```

File: Content.Server/Radiation/Systems/GeigerSystem.cs [hunk 3]
```diff
@@ -152,19 +152,23 @@ private void UpdateSound(EntityUid uid, GeigerComponent? component = null)
...
+
+            if (component.LocalSoundOnly
+                && component.User is not null
+                && _player.TryGetSessionByEntity(component.User.Value, out var session))
+            {
+                component.Stream = _audio.PlayGlobal(sound, session, param)?.Entity;
+                return;
+            }
+            component.Stream = _audio.PlayEntity(sound, Filter.Pvs(uid), uid, true, param)?.Entity;
```
Comment (DEATHB4DEFEAT) [inline Content.Server/Radiation/Systems/GeigerSystem.cs:170]:
```suggestion

            component.Stream = _audio.PlayEntity(sound, Filter.Pvs(uid), uid, true, param)?.Entity;
```

File: Content.Shared/Radiation/Components/GeigerComponent.cs [hunk 4]
```diff
@@ -83,6 +79,31 @@ public sealed partial class GeigerComponent : Component
...
+    ///     Controls how quickly the sound gets quieter as a player moves away from the geiger counter.
+    /// </summary>
+    [DataField]
+    public float RolloffFactorParameters = 4f;
+
+    /// <summary>
+    ///     Controls the maximum distance that players can hear geiger counter ticks.
+    /// </summary>
+    [DataField]
+    public float MaxDistanceParameters = 10f;
```
Comment (DEATHB4DEFEAT) [inline Content.Shared/Radiation/Components/GeigerComponent.cs:106]: AudioParams are serializable, use that instead.
```suggestion
    public float AudioVolume = -4f;

    /// <summary>
    ///     Controls how quickly the sound gets quieter as a player moves away from the geiger counter.
    /// </summary>
    [DataField]
    public float AudioRolloffFactor = 4f;

    /// <summary>
    ///     Controls the maximum distance that players can hear geiger counter ticks.
    /// </summary>
    [DataField]
    public float AudioMaxDistance = 10f;
```

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]

**Comment by VMSolidus** [comment]
@DEATHB4DEFEAT swapped it out for its own component set AudioParameters.

**Review by Peptide90 — APPROVED** [review]

**Pull request merged**